packages/rocketchat-lib/server/models/Rooms.js

```diff
@@ -256,6 +256,19 @@ class ModelRooms extends RocketChat.models._Base {
 		return this._db.find(query, options);
 	}
 
+	findByNameAndTypeAndNotSecret(name, type, options) {
```

Comment:
I noticed that this semantic is not sufficient: Even if a room is secret, it should be listed if the user is a member of this room. Thus, the semantic better should be `findListableByNameAndType`

Reply:
With this change, the private channel -> marked as `secret` will be displayed on the search, if the user is part of the channel and not otherwise.